src/riscv/lib/src/pvm/linux/signals.rs

```diff
@@ -215,12 +229,54 @@ struct_layout! {
 }
 
 impl<MC: MemoryConfig, M: ManagerReadWrite> MachineCoreState<MC, M> {
+    /// Set the hart state to a signal handler
+    pub fn dispatch_signal(&mut self, signal: Signal, handler: VirtAddr) {
+        let restorer = self
+            .push_signal_context(signal)
+            .expect("Signal handling should succeed");
```

Comment:
I'm not sure if you have any control over `push_signal_context` failing or not at this point. Probably best to forward the error.